app/models/import_users.rb

```diff
@@ -0,0 +1,95 @@
+class ImportUsers < Struct.new(:import_id)
+  def perform
+    import = Import.find(import_id)
+    content_hash = JSON.parse(import.upload_data, :symbolize_names => true)
+    total_users_count = content_hash[:users].size
+    batch_size = 500
+    total_batches = (total_users_count/batch_size.to_f).ceil
+    start_index = 0
+    end_index = batch_size - 1
+    importing_portal = ImportingPortal.find(:first, :conditions => {:portal_url => content_hash[:portal_name]})
+    importing_portal = importing_portal || ImportingPortal.create({:portal_url => content_hash[:portal_name]})
+    import.update_attribute(:total_imports, total_users_count)
+    
+    0.upto(total_batches-1){|batch_index|
+      start_index = batch_index * batch_size
+      end_index = (batch_index == total_batches - 1)? (total_users_count - 1) : (start_index + batch_size - 1)
+      ActiveRecord::Base.transaction do
+        start_index.upto(end_index){|index|
+          user = content_hash[:users][index]
+          new_user = User.find_by_email(user[:email])
+          unless new_user
+            new_user = User.find_by_login(user[:login])
+            duplicate_user = nil
+            if new_user
+              new_index = ImportDuplicateUser.find_all_by_login(user[:login]).count + 1
+              user[:login] = "#{user[:login]}-#{new_index}"
+              duplicate_user = ImportDuplicateUser.create({
+                :login => user[:login],
+                :email => user[:email],
+                :duplicate_by => ImportDuplicateUser::DUPLICATE_BY_LOGIN_AND_EMAIL,
+                :data => user,
+                :import_id => import.id
+              })
+            end
+            new_user = User.create({
+              :email => user[:email],
+              :login => user[:login],
+              :last_name => user[:last_name],
+              :first_name => user[:first_name],
+              :password => "password", #default password
+              :password_confirmation => "password",
```

Comment:
Instead of using 'password' here it would be better to use `Devise.friendly_token[0,20]` which generates a random string that is appropriate for a password.